client/pages/exercises.jsx

```diff
@@ -64,18 +70,18 @@ export default function Exercises(props) {
         }
       }
     });
-    const body = { workoutId: 1, exerciseIds: savedExercises, userId: 1 };
+    const body = { workoutId, exerciseIds: savedExercises, userId: user.userId };
+    const accessToken = window.localStorage.getItem('strive-user-info');
     fetch('/api/workout/new-exercises', {
       method: 'POST',
       headers: {
-        'Content-Type': 'application/json'
+        'Content-Type': 'application/json',
+        'X-Access-Token': accessToken
       },
       body: JSON.stringify(body)
     })
-      .then(response => {
-        response.json();
-      })
-      .then(data => clearExercises())
+      .then(response => response.json())
+      .then(result => clearExercises())
       .catch(err => console.error('ERROR:', err));
     window.location.hash = 'workout';
   }
```

Comment:
Here again don't change the hash here, instead wait until your fetch is actually done (`.then()`)